File: app/ios/RunninWatch Watch App/Components/RunninLogo.swift
import SwiftUI

/// Logo header reusada em todas as telas do Watch.
/// Renderiza "RUNNIN.AI" em mono bold — não temos asset SVG no Watch bundle
/// (custo de bundle/coordenação não vale), texto faz o serviço.
///
/// O fundo do ".AI" pega `accentColor` da skin atual do iPhone (vem via
/// WCSession). Default cyan = skin Artico.
struct RunninLogo: View {
    @EnvironmentObject var state: WatchRunState

    var body: some View {
        HStack(spacing: 0) {
            Text("RUNNIN")
                .font(state.scaledFont(size: 11, weight: .heavy))
                .tracking(1.4)
                .foregroundStyle(.white)
            // RoundedRectangle no .background em vez de cor direta — em
            // watchOS 11 o `.background(Color)` com .padding em Text estava
            // expandindo verticalmente pra encher altura do parent VStack
            // do TabView page (visível no screenshot do user: bloco de cor
            // crescendo ~5x). Shape com fill respeita o bounding box do
            // padded text. fixedSize na HStack garante que o conjunto não
            // absorve altura do parent.
            Text(".AI")
                .font(state.scaledFont(size: 9, weight: .bold))
                .padding(.horizontal, 3)
                .padding(.vertical, 1)
                .foregroundStyle(.black)
                .background(
                    RoundedRectangle(cornerRadius: 2)
                        .fill(state.accentColor)
                )
        }
        .fixedSize()
    }
}

File: app/android/wear/src/main/kotlin/com/s6lab/runnin/wear/components/RunninLogo.kt
package com.s6lab.runnin.wear.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.snapshots.SnapshotStateObserver
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextDecoration
import androidx.compose.ui.unit.dp
import androidx.wear.compose.material.Text
import com.s6lab.runnin.wear.models.WatchRunState
import com.s6lab.runnin.wear.theme.scaledFont

/**
 * Logo header reusada em todas as telas do Watch.
 * "RUNNIN" branco bold + ".AI" preto sobre quadrado accentColor.
 *
 * Paralelo do `RunninLogo` do Apple Watch (Components/RunninLogo.swift).
 */
@Composable
fun RunninLogo() {
    val state = WatchRunState.shared
    val accent: Color = state.accentColor
    Row {
        Text(
            text = "RUNNIN",
            color = Color.White,
            style = scaledFont(11f, FontWeight.Black),
        )
        Text(
            text = ".AI",
            color = Color.Black,
            style = scaledFont(9f, FontWeight.Bold),
            modifier = Modifier
                .clip(RoundedCornerShape(2.dp))
                .background(accent)
                .padding(horizontal = 3.dp, vertical = 1.dp),
        )
    }
}
